targets/TARGET_STM/TARGET_STM32F7/TARGET_STM32F746xG/device/TOOLCHAIN_ARM_MICRO/stm32f746xg.sct

```diff
@@ -28,55 +28,46 @@
 ; OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
+; predefine values from pack manager:
+; MBED_ROM_START =0x8000000  MBED_ROM_SIZE =0x100000 (1024 KB)
+; MBED_RAM_START =0x20010000 MBED_RAM_SIZE =0x40000 (256 KB)
+; MBED_RAM1_START=0x20000000 MBED_RAM1_SIZE=0x10000 (64 KB)
+
 #if !defined(MBED_APP_START)
-  #define MBED_APP_START            0x08000000
+  #define MBED_APP_START MBED_ROM_START
 #endif
 
-; STM32F746NG: 1024 KB FLASH (0x100000)
 #if !defined(MBED_APP_SIZE)
-  #define MBED_APP_SIZE             0x100000
-#endif
-
-; 320 KB SRAM (0x50000)
-#if !defined(MBED_RAM_START)
-  #define MBED_RAM_START            0x20000000
-#endif
-
-#if !defined(MBED_RAM_SIZE)
-  #define MBED_RAM_SIZE             0x00050000
+  #define MBED_APP_SIZE MBED_ROM_SIZE
 #endif
 
-
-#if !defined(MBED_BOOT_STACK_SIZE)
-  #define MBED_BOOT_STACK_SIZE      0x400
-#endif
-
-; Total: 114 vectors = 456 bytes (0x1C8) to be reserved in RAM
-#define VECTOR_SIZE                 0x1C8
+#define MBED_VECTTABLE_RAM_START    (MBED_RAM1_START)
+#define NVIC_NUM_VECTORS 114
+#define MBED_VECTTABLE_RAM_SIZE     (((NVIC_NUM_VECTORS * 4) + 7) AND ~7)
+#define MBED_CRASH_REPORT_RAM_START (MBED_VECTTABLE_RAM_START + MBED_VECTTABLE_RAM_SIZE)
 #define MBED_CRASH_REPORT_RAM_SIZE  0x100
-#define MBED_IRAM1_START             (MBED_RAM_START + VECTOR_SIZE + MBED_CRASH_REPORT_RAM_SIZE)
-#define MBED_IRAM1_SIZE              (MBED_RAM_SIZE - VECTOR_SIZE - MBED_CRASH_REPORT_RAM_SIZE)
+#define MBED_RAM0_START             (MBED_CRASH_REPORT_RAM_START + MBED_CRASH_REPORT_RAM_SIZE)
+#define MBED_RAM0_SIZE              (MBED_RAM1_SIZE + MBED_RAM_SIZE - MBED_VECTTABLE_RAM_SIZE - MBED_CRASH_REPORT_RAM_SIZE)
```

Comment:
```suggestion
#define MBED_RAM0_SIZE              (MBED_RAM1_SIZE - MBED_VECTTABLE_RAM_SIZE - MBED_CRASH_REPORT_RAM_SIZE)
```

Reply:
RAM1 and RAM0 are physically separate regions, I am not sure if we can use them as contiguous single region, was this tested in various scenarios.

As per my understanding here below is the memory map:

![image](https://user-images.githubusercontent.com/28542450/55835127-fddffd80-5ae0-11e9-843f-a9cf6dc03b16.png)

Will like to know if case 1 and 2 is tested somehow, 
1. RW_IRAM1 if ends before RAM_START, heap access across RAM and RAM1 is successful 
2. RW_IARM1 spans across RAM boundary

Reply:
Hi

I don't agree with your suggestion...
Solution could be to start RAM0 at RAM?

Information from STM32F746 Reference Manual:
- SRAM1    mapped at 0x20010000 size 0x3C000
– SRAM2    mapped at 0x2004C000 size  0x4000
– DTCM-RAM mapped at 0x20000000 size 0x10000

Information from STM32F7xx pdsc file:
- IRAM1    mapped at 0x20010000 size 0x40000 ==> RAM in mbed-os pack manager
– IRAM2    mapped at 0x20000000 size 0x10000 ==> RAM1 in mbed-os pack manager